classes/class-p4-post-campaign.php

```diff
@@ -315,98 +315,163 @@ private static function campaign_field(
+			$css_vars = self::replace_font_aliases( $css_vars );
 
-			$footer_theme = $meta['campaign_footer_theme'] ?? null;
+			return $css_vars;
+		}
 
-			if ( 'white' === $footer_theme ) {
-				$default_footer_links_color = $meta['campaign_nav_color'] ? $meta['campaign_nav_color'] : '#1A1A1A';
-				$footer_links_color         = $meta['footer_links_color'] ? $meta['footer_links_color'] : $default_footer_links_color;
-				$footer_color               = '#FFFFFF';
-			} else {
-				switch ( ( $meta['campaign_logo_color'] ?? null ) ) {
-					case 'dark':
-						$footer_links_color = '#1A1A1A';
-						break;
-					case 'green':
-						$footer_links_color = '#2caf4e';
-						break;
-					default:
-						$footer_links_color = '#FFFFFF';
-				}
+		/**
+		 * DEPRECATE: This function replaces a font names in the CSS variables.
+		 *
+		 * @param array $css_vars The array containing the CSS variables.
+		 * @return array The variables for the footer theme.
+		 */
+		public static function replace_font_aliases( array $css_vars ): array {
+			// TODO: Remove these special cases.
+			if ( isset( $css_vars['campaign_header_primary'] ) ) {
+				$css_vars['campaign_header_primary'] = str_replace( 'Montserrat_Light', 'Montserrat', $css_vars['campaign_header_primary'] );
+			}
 
-				$footer_color = $meta['campaign_nav_color'] ?? null;
+			if ( isset( $css_vars['campaign_body_font'] ) ) {
+				$css_vars['campaign_body_font'] = str_replace( 'Montserrat_Light', 'Montserrat', $css_vars['campaign_body_font'] );
 			}
 
+			return $css_vars;
+		}
+
+		/**
+		 * DEPRECATE: Get the passive button color based on the meta settings.
+		 *
+		 * @param array $meta The meta containing the style settings.
+		 * @return array The variables for the passive button.
+		 */
+		public static function get_passive_button_color( array $meta, array $defaults ): array {
+			// TODO: Remove this "Passive" color map based on hovers.
 			$passive_button_colors_map = [
+				'#ffd204' => '#f36d3a',
 				'#ffd204' => '#ffe467',
 				'#66cc00' => '#66cc00',
 				'#6ed961' => '#a7e021',
 				'#21cbca' => '#77ebe0',
-				'#ee562d' => '#f36d3a',
 				'#7a1805' => '#a01604',
 				'#2077bf' => '#2077bf',
 				'#1b4a1b' => '#1b4a1b',
 			];
 
-			if ( 'minimal' === $meta['campaign_nav_type'] && isset( $meta['campaign_nav_color'] ) ) {
-				$nav_color = $meta['campaign_nav_color'];
+			$css_vars = [];
+
+			$css_vars['passive_button_color'] = isset( $meta['campaign_primary_color'] ) && $meta['campaign_primary_color']
+			? $passive_button_colors_map[ strtolower( $meta['campaign_primary_color'] ) ]
+			: '#f36d3a';
+
+			return $css_vars;
+		}
+
+		/**
+		 * DEPRECATE: Replaces the legacy mapping to add missing or composed variables,
+		 * returns an iterable flat set of variables.
+		 *
+		 * @param array $css_vars The array containing the CSS variables.
+		 * @return array The variables for the footer theme.
+		 */
+		public static function transform_legacy_mapping( array $css_vars ): array {
```

Comment:
As these variables may not represent the new scheme, I'll say stop using them at this point (before rendering), so we can transform the meta values accordingly if they are old, but render new names with a polished scheme since now, and use those in the SCSS files.